### PR TITLE
[ios] Optimize event uploading

### PR DESCRIPTION
**Pull request by boundsj**
Upload any events collected when the app was in use for apps that only have "when in use" location permissions. Also, for all apps, avoid posting event data if there is only a single event.

### CODE REVIEW

File: platform/ios/src/MGLMapboxEvents.m [hunk 1]
```diff
@@ -489,6 +504,9 @@ - (void)postEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)events {
             }
             [strongSelf pushDebugEvent:MGLEventTypeLocalDebug withAttributes:@{MGLEventKeyLocalDebugDescription: @"post",
                                                                                     @"debug.eventsCount": @(events.count)}];
+            
+            [[UIApplication sharedApplication] endBackgroundTask:_backgroundTaskIdentifier];
+            _backgroundTaskIdentifier = UIBackgroundTaskInvalid;
```
Comment (friedbunny) [inline platform/ios/src/MGLMapboxEvents.m]: It looks like the background task won’t be invalidated if there was a network error (see [L503](https://github.com/mapbox/mapbox-gl-native/pull/6737/files#diff-6e8e31cd0ffe52aeda9c8b2671bcea19R503)), which would mean the app continues on in the background until it’s killed.

Comment (boundsj) [inline platform/ios/src/MGLMapboxEvents.m]: Thanks! Fixed 7688bcc

**Comment by friedbunny** [comment]
This was originally based on the current `master`, but the `release-ios-v3.4.0` is one Android commit behind that — could you rebase on the release branch?

File: platform/ios/src/MGLMapboxEvents.m [hunk 2]
```diff
@@ -485,10 +500,15 @@ - (void)postEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)events {
             if (error) {
                 [strongSelf pushDebugEvent:MGLEventTypeLocalDebug withAttributes:@{MGLEventKeyLocalDebugDescription: @"Network error",
                                                                                         @"error": error}];
+                [[UIApplication sharedApplication] endBackgroundTask:_backgroundTaskIdentifier];
+                _backgroundTaskIdentifier = UIBackgroundTaskInvalid;
                 return;
```
Comment (friedbunny) [inline platform/ios/src/MGLMapboxEvents.m]: If we removed this `return` and then enclosed the subsequent `pushDebugEvent` in an `else` statement, I think we could avoid duplicating the background task invalidation.

Comment (boundsj) [inline platform/ios/src/MGLMapboxEvents.m]: Thanks again. d648594 refactors as you described and squashes this all into a single commit on top of the release branch.

**Pull request merged**